lib/seadsa/DsaLibFuncInfo.cc

```diff
@@ -33,19 +44,43 @@ void DsaLibFuncInfo::initialize() const {
   using namespace llvm::sys::fs;
   using namespace llvm;
 
-  m_awi = &getAnalysis<AllocWrapInfo>();
-  m_tliWrapper = &getAnalysis<TargetLibraryInfoWrapperPass>();
-
   if (isInitialized || XSpecFiles.empty()) return;
   isInitialized = true;
 
+  m_awi = &getAnalysis<AllocWrapInfo>();
+  m_tliWrapper = &getAnalysis<TargetLibraryInfoWrapperPass>();
+
   SMDiagnostic err;
 
+  auto exePath = getMainExecutable(nullptr, nullptr);
+  StringRef exeDir = llvm::sys::path::parent_path(exePath);
+
+  if (XUseClibSpec) {
+    if (XUse32BitSpec) {
+      const StringRef libcSpecRel = "../lib/libc-32.spec.bc";
+      llvm::SmallString<256> libcSpecAbs = libcSpecRel;
+      make_absolute(exeDir, libcSpecAbs);
+
+      m_modules.push_back(parseIRFile(libcSpecAbs.str(), err, m_ctx));
+    } else {
+      const StringRef libcSpecRel = "../lib/libc.spec.bc";
```

Comment:
probably good idea to embed architecture in both spec files. We do sometimes run things in a 32 bit container, in which case, the architecture is assumed to be 32 bit by default.